samples/linker-instructions-advanced.md

```diff
@@ -0,0 +1,63 @@
+
+## Using the tasks package
+
+The linker tasks package ([ILLink.Tasks](https://dotnet.myget.org/feed/dotnet-core/package/nuget/Illink.Tasks)) contains MSBuild targets that become a part of the referencing project's build, using the mechanism described in the nuget docs.
+
+When this package is referenced in a project, the publish target is augmented with additional targets that run the linker on the app's assemblies before they are placed in the publish directory.
+
+These targets will run the linker on all managed assemblies that are a part of the app, including dependencies from project references and package references. The linker will attempt to determine which parts of the code (in the project and its dependencies) are unnecessary, and it will remove assemblies or parts of assemblies that it determines to be safe to remove. By default this behavior is fairly conservative: the linker will always keep code in the application and its non-framework dependencies, only removing unused parts of the framework assemblies (this may change in the future as we improve the linker's heuristics).
+
+Even with the current conservative behavior, there may be cases in which the linker removes code that the application expects to be present at runtime. For example, the application may use reflection to load and call code at runtime, and the linker will not be able to catch these cases perfectly. To explicitly tell the linker to keep certain code in the linked output, it is possible to specify additional roots via MSBuild properties and xml root descriptor files.
+
+Root assemblies can be specified with the LinkerRootAssemblies ItemGroup:
+
+```xml
+<ItemGroup>
+  <LinkerRootAssemblies Include="MyAssembly" />
+</ItemGroup>
+```
+
+This ItemGroup should contain the logical names of assemblies, not the filenames (so the assembly names should not have extensions).
+
+The linker roots can also be specified at a more granular level using xml root descriptor files, whose format is documented in the mono/linker repo. These files should be specified in the LinkerRootDescriptors ItemGroup:
+
+```xml
+<ItemGroup>
+  <LinkerRootDescriptors Include="path/to/rootDescriptor.xml" />
+</ItemGroup>
+```
```

Comment:
For the above cases, would an example about that illustrated the "root whole assembly" vs the "add particular field/method/type".   This would help answer the "what do I do when something goes wrong" question.